<!doctype html>







<html
  class="not-ready lg:text-base"
  style="--bg:#faf8f1"
  lang="en-us"
  dir="ltr"
><head>
  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <meta
    name="viewport"
    content="width=device-width, initial-scale=1, shrink-to-fit=no"
  />

  
  <title>Centering and Scaling Before PCA in Single Cell RNA-seq; Why It Matters - Bioinformatics Blogs By Nasim</title>

  
  <meta name="theme-color" />

  <meta name="description" content="üß† Centering and Scaling Before PCA; Why It Matters
When performing PCA on gene expression data, centering and scaling aren‚Äôt just optional, they‚Äôre essential to ensure PCA captures true biological variation rather than numerical artifacts. PCA is powerful, but only when the data are centered and scaled correctly.
Otherwise, it captures the wrong kind of ‚Äúvariance.‚Äù
üéØ Centering
PCA ideally looks for directions of maximum variance around the mean of each gene.
But if we don‚Äôt center the data, PCA instead measures variance around the origin (0), not around each gene‚Äôs mean." />
  <meta name="author" content="Nasim" /><link rel="preload stylesheet" as="style" href="https://nasimbio.github.io/bioinfo-blog/main.min.css" />

  
  <link rel="preload" as="image" href="https://nasimbio.github.io/bioinfo-blog/theme.png" />

  <link rel="preload" as="image" href="https://www.gravatar.com/avatar/5d163c4d9d2907703e25364dcb8ef63e?s=160&amp;d=identicon" />

  <link rel="preload" as="image" href="https://nasimbio.github.io/bioinfo-blog/github.svg" /><link rel="preload" as="image" href="https://nasimbio.github.io/bioinfo-blog/linkedin.svg" />

  <script
    defer
    src="https://nasimbio.github.io/bioinfo-blog/highlight.min.js"
    onload="hljs.initHighlightingOnLoad();"
  ></script>

  
  <link
    rel="icon"
    href="https://nasimbio.github.io/bioinfo-blog/favicon.ico"
  />
  <link
    rel="apple-touch-icon"
    href="https://nasimbio.github.io/bioinfo-blog/apple-touch-icon.png"
  />

  <meta name="generator" content="Hugo 0.152.2">
  <meta itemprop="name" content="Centering and Scaling Before PCA in Single Cell RNA-seq; Why It Matters">
  <meta itemprop="description" content="üß† Centering and Scaling Before PCA; Why It Matters
When performing PCA on gene expression data, centering and scaling aren‚Äôt just optional, they‚Äôre essential to ensure PCA captures true biological variation rather than numerical artifacts. PCA is powerful, but only when the data are centered and scaled correctly. Otherwise, it captures the wrong kind of ‚Äúvariance.‚Äù
üéØ Centering
PCA ideally looks for directions of maximum variance around the mean of each gene. But if we don‚Äôt center the data, PCA instead measures variance around the origin (0), not around each gene‚Äôs mean.">
  <meta itemprop="datePublished" content="2025-11-13T00:00:00+00:00">
  <meta itemprop="dateModified" content="2025-11-13T00:00:00+00:00">
  <meta itemprop="wordCount" content="386">
  <meta itemprop="keywords" content="Single-Cell,Seurat,Centering,Scaling,PCA,Heatmap"><meta property="og:url" content="https://nasimbio.github.io/bioinfo-blog/post/centering-and-scaling-before-pca-why-it-matters/">
  <meta property="og:site_name" content="Bioinformatics Blogs By Nasim">
  <meta property="og:title" content="Centering and Scaling Before PCA in Single Cell RNA-seq; Why It Matters">
  <meta property="og:description" content="üß† Centering and Scaling Before PCA; Why It Matters
When performing PCA on gene expression data, centering and scaling aren‚Äôt just optional, they‚Äôre essential to ensure PCA captures true biological variation rather than numerical artifacts. PCA is powerful, but only when the data are centered and scaled correctly. Otherwise, it captures the wrong kind of ‚Äúvariance.‚Äù
üéØ Centering
PCA ideally looks for directions of maximum variance around the mean of each gene. But if we don‚Äôt center the data, PCA instead measures variance around the origin (0), not around each gene‚Äôs mean.">
  <meta property="og:locale" content="en_us">
  <meta property="og:type" content="article">
    <meta property="article:section" content="post">
    <meta property="article:published_time" content="2025-11-13T00:00:00+00:00">
    <meta property="article:modified_time" content="2025-11-13T00:00:00+00:00">
    <meta property="article:tag" content="Single-Cell">
    <meta property="article:tag" content="Seurat">
    <meta property="article:tag" content="Centering">
    <meta property="article:tag" content="Scaling">
    <meta property="article:tag" content="PCA">
    <meta property="article:tag" content="Heatmap">

  <meta name="twitter:card" content="summary">
  <meta name="twitter:title" content="Centering and Scaling Before PCA in Single Cell RNA-seq; Why It Matters">
  <meta name="twitter:description" content="üß† Centering and Scaling Before PCA; Why It Matters
When performing PCA on gene expression data, centering and scaling aren‚Äôt just optional, they‚Äôre essential to ensure PCA captures true biological variation rather than numerical artifacts. PCA is powerful, but only when the data are centered and scaled correctly. Otherwise, it captures the wrong kind of ‚Äúvariance.‚Äù
üéØ Centering
PCA ideally looks for directions of maximum variance around the mean of each gene. But if we don‚Äôt center the data, PCA instead measures variance around the origin (0), not around each gene‚Äôs mean.">

  <link rel="canonical" href="https://nasimbio.github.io/bioinfo-blog/post/centering-and-scaling-before-pca-why-it-matters/" />
</head>
<body
    class="bg-(--bg) text-black antialiased duration-200 ease-out [-webkit-tap-highlight-color:transparent] dark:text-white"
  ><header
  class="mx-auto flex h-[4.5rem] max-w-(--w) px-8 whitespace-nowrap lg:justify-center"
>
  <div class="relative z-50 flex items-center ltr:mr-auto rtl:ml-auto">
    <a
      class="-translate-y-[1px] text-2xl font-medium"
      href="https://nasimbio.github.io/bioinfo-blog/"
      >Bioinformatics Blogs By Nasim</a
    >
    <div
      class="btn-dark text-[0px] ltr:ml-4 rtl:mr-4 h-6 w-6 shrink-0 cursor-pointer [background:url(./theme.png)_left_center/_auto_theme('spacing.6')_no-repeat] [transition:_background-position_0.4s_steps(5)] dark:[background-position:right]"
      role="button"
      aria-label="Dark"
    ></div>
  </div>

  <div
    class="btn-menu relative z-50 flex h-[4.5rem] w-[5rem] shrink-0 cursor-pointer flex-col items-center justify-center gap-2.5 lg:hidden ltr:-mr-8 rtl:-ml-8"
    role="button"
    aria-label="Menu"
  ></div>

  <script>
    
    const htmlClass = document.documentElement.classList;
    setTimeout(() => {
      htmlClass.remove('not-ready');
    }, 10);

    
    const btnMenu = document.querySelector('.btn-menu');
    btnMenu.addEventListener('click', () => {
      htmlClass.toggle('open');
    });

    
    const metaTheme = document.querySelector('meta[name="theme-color"]');
    const lightBg = '#faf8f1'.replace(/"/g, '');
    const setDark = (isDark) => {
      metaTheme.setAttribute('content', isDark ? '#000' : lightBg);
      htmlClass[isDark ? 'add' : 'remove']('dark');
      localStorage.setItem('dark', isDark);
    };

    
    const darkScheme = window.matchMedia('(prefers-color-scheme: dark)');
    if (htmlClass.contains('dark')) {
      setDark(true);
    } else {
      const darkVal = localStorage.getItem('dark');
      setDark(darkVal ? darkVal === 'true' : darkScheme.matches);
    }

    
    darkScheme.addEventListener('change', (event) => {
      setDark(event.matches);
    });

    
    const btnDark = document.querySelector('.btn-dark');
    btnDark.addEventListener('click', () => {
      setDark(localStorage.getItem('dark') !== 'true');
    });
  </script>

  <div
    class="nav-wrapper fixed inset-x-0 top-full z-40 flex h-full flex-col justify-center bg-(--bg) pb-16 duration-200 select-none lg:static lg:h-auto lg:flex-row lg:bg-transparent! lg:pb-0 lg:transition-none"
  ><nav
      class="mt-12 flex justify-center space-x-10 lg:mt-0 lg:items-center ltr:lg:ml-14 rtl:space-x-reverse rtl:lg:mr-14 dark:invert"
    >
      <a
        class="h-7 w-7 text-[0px] [background:var(--url)_center_center/cover_no-repeat] lg:h-6 lg:w-6"
        style="--url: url(./github.svg)"
        href="https://github.com/nasimbio"
        target="_blank"
        rel="me"
      >github</a>
      <a
        class="h-7 w-7 text-[0px] [background:var(--url)_center_center/cover_no-repeat] lg:h-6 lg:w-6"
        style="--url: url(./linkedin.svg)"
        href="https://linkedin.com/in/nasimrahmatpour"
        target="_blank"
        rel="me"
      >linkedin</a>
    </nav>
  </div>
</header>
<main
      class="prose prose-neutral dark:prose-invert relative mx-auto min-h-[calc(100vh-9rem)] max-w-(--w) px-8 pt-14 pb-16"
    ><article>
  <header class="mb-14">
    <h1 class="my-0! pb-2.5">Centering and Scaling Before PCA in Single Cell RNA-seq; Why It Matters</h1><div class="text-xs antialiased opacity-60"><time>Nov 13, 2025</time><span class="mx-1">&middot;</span>
      <span>Nasim</span></div></header>

  <section><p>üß† Centering and Scaling Before PCA; Why It Matters</p>
<p>When performing PCA on gene expression data, centering and scaling aren‚Äôt just optional, they‚Äôre essential to ensure PCA captures true biological variation rather than numerical artifacts. PCA is powerful, but only when the data are centered and scaled correctly.
Otherwise, it captures the wrong kind of ‚Äúvariance.‚Äù</p>
<p>üéØ Centering</p>
<p>PCA ideally looks for directions of maximum variance around the mean of each gene.
But if we don‚Äôt center the data, PCA instead measures variance around the origin (0), not around each gene‚Äôs mean.</p>
<p>This subtle difference matters a lot:</p>
<p>That means housekeeping genes which are highly expressed everywhere even if they barely vary appear to contribute huge variance simply because their mean values are far from zero dominate the first PC, even though they don‚Äôt separate cell types.
The result: the first principal component becomes dominated by highly expressed, low-variance genes, masking real biological structure.</p>
<p>üß© High mean ‚â† informative variance</p>
<p>‚úÖ Centering removes mean bias so PCA focuses on true biological variation.</p>
<p>‚öñÔ∏è Scaling</p>
<p>If we skip scaling, genes with large numeric variance (often noisy or highly expressed) overpower the analysis. Z-scoring gives every gene equal influence, so PCA highlights correlated expression patterns not raw magnitude.</p>
<p>üß© Scaling removes numeric bias, not biological signal.</p>
<p>üî¨ In Seurat</p>
<p>1Ô∏è‚É£ FindVariableFeatures() ‚Üí picks biologically variable genes</p>
<p>2Ô∏è‚É£ ScaleData() ‚Üí z-score (center + scale)</p>
<p>3Ô∏è‚É£ RunPCA() ‚Üí captures real structure across cells</p>
<p>üé® Same reason we z-score heatmaps:</p>
<p>To compare genes relative to their own mean and variance.</p>
<p>DoHeatmap() in Seurat already uses scaled data.</p>
<p>Summary:</p>
<p>PCA without centering/scaling shows who‚Äôs loudest.
PCA with centering/scaling shows who‚Äôs different.</p>
<p>üß¨ Bulk RNA-seq vs. Single-cell RNA-seq: Why Scaling Decisions Differ</p>
<p>Whether you scale before PCA depends on type of data. In bulk RNA each sample already represent an average over many cells and variance across samples reflect biological conditions. So, for bulk RNA you usually don‚Äôt scale before PCA since absolute variance reflects biological meaning and it corresponds to PCA based on covariance. For single cell, you should scale before PCA otherwise PCA will be driven by highly expressed genes rather than meaningful biological expression. This corresponds to PCA based on correlation matrix.</p>
<p>To learn more, check out this excellent explanation on ‚Äúchatomics‚Äù, YouTube channel. I always learn from him.</p>
<p><a href="https://www.youtube.com/watch?v=P7kj0GLTgS4">https://www.youtube.com/watch?v=P7kj0GLTgS4</a></p>
</section>

  <footer class="mt-12 flex flex-wrap"><a
      class="mb-1.5 rounded-lg bg-black/[3%] px-5 py-1 no-underline hover:bg-black/[6%] ltr:mr-1.5 rtl:ml-1.5 dark:bg-white/[8%] dark:hover:bg-white/[12%]"
      href="https://nasimbio.github.io/bioinfo-blog/tags/single-cell"
      >single-cell</a
    ><a
      class="mb-1.5 rounded-lg bg-black/[3%] px-5 py-1 no-underline hover:bg-black/[6%] ltr:mr-1.5 rtl:ml-1.5 dark:bg-white/[8%] dark:hover:bg-white/[12%]"
      href="https://nasimbio.github.io/bioinfo-blog/tags/seurat"
      >Seurat</a
    ><a
      class="mb-1.5 rounded-lg bg-black/[3%] px-5 py-1 no-underline hover:bg-black/[6%] ltr:mr-1.5 rtl:ml-1.5 dark:bg-white/[8%] dark:hover:bg-white/[12%]"
      href="https://nasimbio.github.io/bioinfo-blog/tags/centering"
      >Centering</a
    ><a
      class="mb-1.5 rounded-lg bg-black/[3%] px-5 py-1 no-underline hover:bg-black/[6%] ltr:mr-1.5 rtl:ml-1.5 dark:bg-white/[8%] dark:hover:bg-white/[12%]"
      href="https://nasimbio.github.io/bioinfo-blog/tags/scaling"
      >Scaling</a
    ><a
      class="mb-1.5 rounded-lg bg-black/[3%] px-5 py-1 no-underline hover:bg-black/[6%] ltr:mr-1.5 rtl:ml-1.5 dark:bg-white/[8%] dark:hover:bg-white/[12%]"
      href="https://nasimbio.github.io/bioinfo-blog/tags/pca"
      >PCA</a
    ><a
      class="mb-1.5 rounded-lg bg-black/[3%] px-5 py-1 no-underline hover:bg-black/[6%] ltr:mr-1.5 rtl:ml-1.5 dark:bg-white/[8%] dark:hover:bg-white/[12%]"
      href="https://nasimbio.github.io/bioinfo-blog/tags/heatmap"
      >heatmap</a
    ></footer><nav
    class="mt-24 flex overflow-hidden rounded-xl bg-black/[3%] text-lg leading-[1.2]! *:flex *:w-1/2 *:items-center *:p-5 *:font-medium *:no-underline dark:bg-white/[8%] [&>*:hover]:bg-black/[2%] dark:[&>*:hover]:bg-white/[3%]"
  ><a
      class="justify-end pl-3 ltr:ml-auto rtl:mr-auto"
      href="https://nasimbio.github.io/bioinfo-blog/post/single-cell-data-analysis-understanding-cca-rpca-pca-and-mapquery-in-seurat-how-they-relate-and-differ/"
      ><span>Single-Cell Data Analysis: Understanding CCA, RPCA, PCA, and MapQuery in Seurat; How They Relate and Differ</span><span class="ltr:ml-1.5 rtl:mr-1.5">‚Üí</span></a
    ></nav></article></main><footer
  class="mx-auto flex h-[4.5rem] max-w-(--w) items-center px-8 text-xs tracking-wider uppercase opacity-60"
>
  <div class="mr-auto">&copy;2025
    <a class="link" href="https://nasimbio.github.io/bioinfo-blog/">Bioinformatics Blogs By Nasim</a></div>
  <a class="link mx-6" href="https://gohugo.io/" rel="noopener" target="_blank"
    >powered by hugoÔ∏èÔ∏è</a
  >Ô∏è
  <a
    class="link"
    href="https://github.com/nanxiaobei/hugo-paper"
    rel="noopener"
    target="_blank"
    >hugo-paper</a
  >
</footer>
</body>
</html>
